packages/website/src/server/oauth/core/oauth2.ts

```diff
@@ -115,14 +116,19 @@ export async function callback ({ url, params, cookies, site }: APIContext, oaut
 		challenges.delete(fullState)
 	}
 
+	const headers: HeadersInit = {
+		Accept: 'application/json',
+		'Content-Type': 'application/x-www-form-urlencoded',
+		'User-Agent': 'PronounDB Authentication Agent/2.0 (+https://pronoundb.org)',
+	}
+
+	if (!oauth.oauthNoAuthorizationHeader) {
+		headers.Authorization = `Basic ${Buffer.from(`${oauth.clientId}:${oauth.clientSecret}`).toString('base64')}`
+	}
```

Comment:
Does SourceHut explicitly complain about the presence of the basic authentication? I kinda liked how this "mini oauth lib" I've been using in many projects of mine over the years never had to accommodate for things like this...

Reply:
Yeah, it throws an error that it cannot accept both `Authorization` header and `client_id`//`client_secret`

Reply:
Uff, that sucks but understandable. Complying with [RFC 6749 § 2.3.1](https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1), it'd be advisable to drop the inclusion of the credentials within the body instead.

It'd be ideal to always drop it unless dealing with a non-compliant OAuth2 server implementation, but for the sake of simplicity it can be kept as an opt-out parameter like this, rather than opt-in (except if you're feeling like manually reviewing which authorization servers are being annoying about it 😄)

Reply:
Applied the suggestions c:

(Also no, I'm lazy so keeping this as-is should be okay, ig?)